include/linker/ramfunc.ld

```diff
@@ -0,0 +1,19 @@
+SECTION_DATA_PROLOGUE(.ramfunc,,)
+{
+#ifdef CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT
+	. = ALIGN( 1 << LOG2CEIL(__ramfunc_ram_end - __ramfunc_ram_start));
+#endif
```

Comment:
@SebastianBoe @andrewboie could you review the linker script?
Basically we need to satisfy
 - ARMv7M MPU with requirement for power-of-two alignment
- ARMv8-M with requirement for non-overlapping regions (aligned with 32-byte minimum)